mirgecom/simutil.py

```diff
@@ -434,6 +435,21 @@ def create_parallel_grid(comm, generate_grid):
     return generate_and_distribute_mesh(comm=comm, generate_mesh=generate_grid)
 
 
+def get_reasonable_memory_pool(ctx, queue):
+    """Return an SVM or buffer memory pool based on what the device supports."""
+    from pyopencl.characterize import has_coarse_grain_buffer_svm
+    import pyopencl.tools as cl_tools
+
+    if has_coarse_grain_buffer_svm(queue.device) and hasattr(cl_tools, "SVMPool"):
+        logger.info("Using SVM-based memory pool")
+        return cl_tools.SVMPool(cl_tools.SVMAllocator(  # pylint: disable=no-member
+            ctx, alignment=0, queue=queue))
+    else:
+        from warnings import warn
+        warn("No SVM support, returning a CL buffer-based memory pool")
+        return cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue))
```

Comment:
Why `simutils` and not _pyopencl_ ?

Reply:
My thought was that this function is very specific to our use case - other applications using pyopencl probably won't care much for this function since it is unlikely they would prefer an SVM pool over a buffer pool. I thought about putting this in grudge, but it also doesn't feel relevant there.

Reply:
I defer to your judgement on this, i was mostly curious